README.md

```diff
@@ -19,14 +18,55 @@ Once you have implemented the `sa` program (and tested it to the best of your ab
 
 *Which algorithm did you use to construct the suffix array. What is its running time?*
 
+# ANWSER (Algorithm):
+I used Prefix-doubling.
+The doubling of prefixes (in each iter) gives a max of log(n) iterations and since bucket-sort is O(n) per iteration we have a total running time of O(nlog(n)).
+I haven't implemented the bucket/radix-sort 'yet' so i guess the running time of my implementation is O(nlog(n)2) for now.
+
 ## Insights you may have had while implementing the algorithm
 
+# ANWSER (Insights):
+I found out that SA's are really cleaver, SA-IS is a motherfucker to implement and prefix doubling is a very nice (and simple) little algorithm. 
```

Comment:
True, although there is a bit more to prefix doubling than meets the eye if you want to do it right :)